### PR TITLE
feat: Add route client

### LINKED ISSUE #1031: Add route client

### PR DESCRIPTION
**Pull request by fqutishat**
closes #1031

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/client/route/client.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	routeSvc, ok := svc.(protocolService)
+	if !ok {
+		return nil, errors.New("cast service to route service failed")
+	}
+
+	connectionStore, err := connectionstore.NewConnectionRecorder(ctx)
```
Comment (troyronda) [inline pkg/client/route/client.go]: Do you need R/W access to connections in this protocol?
(or read only.)

Comment (fqutishat) [reply pkg/client/route/client.go]: just read only

Comment (troyronda) [reply pkg/client/route/client.go]: Also L32

Comment (fqutishat) [reply pkg/client/route/client.go]: change it to connection lookup

File: pkg/client/route/client.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	return &Client{
+		routeSvc:         routeSvc,
+		connectionLookup: connectionLookup,
+	}, nil
+}
+
+// SendRequest send route request
+func (c *Client) SendRequest(connectionID string) (string, error) {
```
Comment (troyronda) [inline pkg/client/route/client.go]: We may need to enhance the naming... `SendRequest` seems very generic.

**Pull request merged**